otherlibs/stdune/src/path.ml

```diff
@@ -950,6 +950,11 @@ let as_in_source_tree_exn t =
       "[as_in_source_tree_exn] called on something not in source tree"
       [ ("t", to_dyn t) ]
 
+let as_outside_build_dir : t -> Outside_build_dir.t option = function
+  | In_source_tree s -> Some (In_source_dir s)
+  | External s -> Some (External s)
+  | In_build_dir _ -> None
+
 let as_outside_build_dir_exn : t -> Outside_build_dir.t = function
```

Comment:
this can be implemented in terms of `as_outside_build_dir` now.

Reply:
I followed the implementation of the other `as_`s which also don't implement in terms of eachother.

Reply:
I only looked at `as_in_source_tree_exn` which is right above your function, and it's implemented in terms of the other one.

Reply:
`as_in_source_tree` loses the content of the `In_build_dir` constructor which we include in the code error so we cannot use it for `as_in_source_tree`.